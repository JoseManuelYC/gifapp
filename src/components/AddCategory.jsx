import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputValue = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (evento) => {
    evento.preventDefault();
    const newInputValue = inputValue.trim();
    if (newInputValue.length <= 1) return;

    /*  setCategories([inputValue, ...categories]); */
    /*     setCategories((categories) => [inputValue, ...categories]); */

    onNewCategory(newInputValue);

    setInputValue("");
  };

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Buscar un gif"
        onChange={onInputValue}
        value={inputValue}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
