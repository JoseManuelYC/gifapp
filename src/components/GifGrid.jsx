import PropTypes from "prop-types";

import { useGifFetch } from "../hooks/useGifFetch";

import { GifItem } from "./GifItem";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useGifFetch(category);

  return (
    <div className="subcontainer">
      <h3>{category}</h3>
      {isLoading && <h2>Cargando...</h2>}
      <div className="card-grid">
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </div>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
