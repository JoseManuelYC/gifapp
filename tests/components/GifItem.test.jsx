import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Haciendo pruebas en el GifItem", () => {
  const title = "Residente del Mal";
  const url = "https://residente.evil.com/RE.jpg";
  test("haciendo match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });
  test("debe existir el title y el url en el componente", () => {
    render(<GifItem title={title} url={url} />);

    //expect(screen.getByRole("img").src).toBe(url);   (Otra manera de verlo)

    const { src, alt } = screen.getByRole("img");

    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("Debe de mostrar el titulo en el componente", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
