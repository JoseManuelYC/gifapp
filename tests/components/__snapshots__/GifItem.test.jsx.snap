// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Haciendo pruebas en el GifItem haciendo match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Residente del Mal"
      src="https://residente.evil.com/RE.jpg"
    />
    <p>
      Residente del Mal
    </p>
  </div>
</div>
`;
